paasta_tools/utils.py

```diff
@@ -3076,20 +3078,28 @@ def read_service_instance_names(
     service: str, instance_type: str, cluster: str, soa_dir: str
 ) -> Collection[Tuple[str, str]]:
     instance_list = []
-    conf_file = f"{instance_type}-{cluster}"
+    if instance_type == "spark":
+        conf_file = f"tron-{cluster}"
+    else:
+        conf_file = f"{instance_type}-{cluster}"
     config = service_configuration_lib.read_extra_service_information(
         service,
         conf_file,
         soa_dir=soa_dir,
         deepcopy=False,
     )
     config = filter_templates_from_config(config)
-    if instance_type == "tron":
+    if instance_type == "tron" or instance_type == "spark":
```

Comment:
nit:
```suggestion
    if instance_type in {"tron", "spark"}:
```

(or maybe even extract this set to `TRON_INSTANCE_TYPES`?)